apps/edit/app.lisp

```diff
@@ -76,32 +71,42 @@
 (defun-bind mouse-cursor (mouse_xy)
 	(defq buffer (elem text_buffer current_text))
 	(bind '(ox oy cx cy sx) (elem text_position current_text))
-
 	(defq cursor_xy (list cx cy) char_wh (vdu-char-size vdu) offset_xy (list ox oy))
 	(setq cursor_xy (map + (map / mouse_xy char_wh) offset_xy)
 		cx (elem 0 cursor_xy) cy (elem 1 cursor_xy))
-	(if (>= cy (length buffer)) (setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
+	(if (>= cy (length buffer)) 
+		(setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
 	(if (> cx (length (elem cy buffer)))
 		(setq cx (max (set-sticky) (length (elem cy buffer)))) (setq sx cx))
-
 	(elem-set text_buffer current_text buffer)
 	(elem-set text_position current_text (list ox oy cx cy sx)))
 
 (defun-bind open-buffer (path)
-	(defq index (length text_store) title path buffer (list (join " " (ascii-char 10))) pos (list 0 0 0 0 0))
-	(unless (or (eql path "") (not (catch (file-stream path) t)))
-		(setq buffer (list))
-		(each-line (lambda (_) (push buffer _)) (file-stream path)))
-	(if (eql title "") 
-		(setq title (cat "Untitled-" (str (setq tmp_num (inc tmp_num))))))
-	(push text_store (list index path title buffer pos))
+	(defq i 0 index (length text_store) pos (list 0 0 0 0 0))
+	(if (and (not (some (lambda (_) (eql "/" _)) path)) (not (eql path ""))) 
+		(setq path (cat home_dir path)))
+	(cond
+		((eql path "")
+			(defq title (cat "Untitled-" (str (setq tmp_num (inc tmp_num)))) 
+				buffer (list (join " " (ascii-char 10))))
+			(push text_store (list index path title buffer pos)))
+		((some (lambda (_) (eql path (elem text_path _))) text_store)
+			(while (< i (length text_store)) 
+				(if (eql path (elem text_path (elem i text_store)))
+					(setq index i)) (setq i (inc i))))
+		(t
+			(defq title path buffer (list))
+			(each-line (lambda (_) (push buffer _)) (file-stream path))
+			(push text_store (list index path title buffer pos))))
 	(elem index text_store))
 
 (defun-bind save-buffer (path)
 	(unless (eql path "")
+		(if (not (some (lambda (_) (eql "/" _)) path)) (setq path (cat home_dir path)))
 		(defq save_buffer (join (elem text_buffer current_text) (ascii-char 10)))
 		(save save_buffer path)
-		(elem-set text_path title current_text path)))
+		(elem-set text_title current_text path)
+		(elem-set text_path current_text path)))
 
```

Comment:
(elem-set text_path current_text path)

Tupple indexes use const ? If they really are const.

(elem-set (const text_path) current_text path)

Reply:
Ahh, I thought the const was implied. I will go through and fix with recommendations.